### ISSUE TITLE
Go schema generator

### ISSUE THREAD

**Issue by vearutop** [asyncapi/asyncapi#100]
Hello, recently I've made a AsyncAPI schema generator for Go (MIT license): https://github.com/swaggest/go-asyncapi. 

It uses reflection to translate Go structures in JSON Schema definitions and arrange them in AsyncAPI schema.

Maybe it is worth adding to https://www.asyncapi.com/v1/tooling/codegens.html.

**Comment by fmvilas** [comment]
Nice! Thanks for creating this @vearutop. However, this is a new category of tools. If I understood correctly, your tool doesn't generate code, but instead, it generates the AsyncAPI document from your code, right?

**Comment by vearutop** [comment]
Yes, you are correct, it follows code-first approach to create schema from code.

**Comment by fmvilas** [comment]
Awesome! Thanks for the clarification. I'll add it under a new category as soon as I have time. Thanks!
